modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/PropsValues.java

```diff
@@ -30,163 +35,193 @@
 
 	public static final int MAX_JOURNAL_ARTICLE_VERSION_COUNT =
 		GetterUtil.getInteger(
-			_PROPERTIES.getProperty(
-				PropsKeys.MAX_JOURNAL_ARTICLE_VERSION_COUNT));
+			PropertiesLoader.get(PropsKeys.MAX_JOURNAL_ARTICLE_VERSION_COUNT));
 
 	public static final int MAX_MB_CATEGORY_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_MB_CATEGORY_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_MB_CATEGORY_COUNT));
 
 	public static final int MAX_MB_MESSAGE_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_MB_MESSAGE_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_MB_MESSAGE_COUNT));
 
 	public static final int MAX_MB_THREAD_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_MB_THREAD_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_MB_THREAD_COUNT));
 
 	public static final int MAX_USER_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_USER_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_USER_COUNT));
 
 	public static final int MAX_USER_TO_GROUP_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_USER_TO_GROUP_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_USER_TO_GROUP_COUNT));
 
 	public static final int MAX_WIKI_NODE_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_WIKI_NODE_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_WIKI_NODE_COUNT));
 
 	public static final int MAX_WIKI_PAGE_COMMENT_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(
-			PropsKeys.MAX_WIKI_PAGE_COMMENT_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_WIKI_PAGE_COMMENT_COUNT));
 
 	public static final int MAX_WIKI_PAGE_COUNT = GetterUtil.getInteger(
-		_PROPERTIES.getProperty(PropsKeys.MAX_WIKI_PAGE_COUNT));
+		PropertiesLoader.get(PropsKeys.MAX_WIKI_PAGE_COUNT));
+
+	public static final int OPTIMIZE_BUFFER_SIZE = GetterUtil.getInteger(
+		PropertiesLoader.get(PropsKeys.OPTIMIZE_BUFFER_SIZE));
 
-	public static final String OUTPUT_DIR = _PROPERTIES.getProperty(
+	public static final String OUTPUT_DIR = PropertiesLoader.get(
 		PropsKeys.OUTPUT_DIR);
 
-	public static final String VIRTUAL_HOST_NAME =
-		_PROPERTIES.getProperty(PropsKeys.VIRTUAL_HOST_NAME);
+	public static final boolean OUTPUT_MERGE = GetterUtil.getBoolean(
+		PropertiesLoader.get(PropsKeys.OUTPUT_MERGE));
+
+	public static final String SCRIPT = PropertiesLoader.get(PropsKeys.SCRIPT);
+
+	public static final String VIRTUAL_HOST_NAME = PropertiesLoader.get(
+		PropsKeys.VIRTUAL_HOST_NAME);
+
+	public static String getActualProperties() {
+		StringBundler sb = new StringBundler();
+
+		for (String key : PropertiesLoader._properties.stringPropertyNames()) {
+			if (!key.startsWith("sample.sql")) {
+				continue;
+			}
+
+			String value = PropertiesLoader._properties.getProperty(key);
+
+			sb.append(key);
+			sb.append(StringPool.EQUAL);
+			sb.append(value);
+			sb.append(StringPool.NEW_LINE);
+		}
+
+		return sb.toString();
+	}
+
+	private static class PropertiesLoader {
```

Comment:
Hi @dantewang ,
I can not understand the whole commit, why you are doing this? and why you have to use SortedProperties here? We only need to sort when dump out the properties file, it does not make sense you do it here.

Tina.

Reply:
@tinatian 1) SortedProperties was used from the beginning, I just copied it here. 2) SF wants to put public ones before private ones, but then the public constant can not be initialized because the private _properties hasn't been initialized. In benchmark project the PropsValues is excluded from SF sorting, but I don't want to exclude it from SF here.